.github/workflows/validate_python.yml

```diff
@@ -0,0 +1,39 @@
+name: Validate Python Code
+
+on:
+  push:
+    branches:
+      - main
+      - 'releases/**'
+  pull_request:
+    types: [opened, synchronize, reopened]
+  workflow_dispatch:
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v2
+
+    - name: Setup python3
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install build dependencies
+      run: python -m pip install -r python_build_requirements.txt
+
+    - name: Validate python style with black
+      run: |
+        black source/codegen examples --check
+
+    - name: Install poetry for validate_examples.py
+      uses: abatilo/actions-poetry@v2.0.0
+      with:
+        poetry-version: 1.1.5
+
+    - name: Validate RFmx and XNET examples
+      run: |
+        python source/codegen/validate_examples.py -p *rfmx*
+        python source/codegen/validate_examples.py -p *xnet*
```

Comment:
We probably oughta put the parameter in quotes, in case some file is added to the root directory that matches the pattern.